### ISSUE TITLE
Masking "reshape" rule not found.

### ISSUE THREAD

**Issue by sschoenholz** [google/jax#2565]
With JAX 0.1.62 the following code:

```python
from jax.experimental import stax
from jax.api import mask, vmap
from jax import random
import jax.numpy as np

init_fn, fn = stax.Dense(1024)

key = random.PRNGKey(0)
_, params = init_fn(key, (-1, 3))

xs = np.array([[1, 2, 3], [4, 5, 6]])

def single_fn(xs):
  return fn(params, np.reshape(xs, (1,) + xs.shape))[0]

print(mask(single_fn, ['n'], '_')([xs[1]], dict(n=2)))
```

Produces the following error: https://pastebin.com/fiQ9Ux5d

If I am doing something wrong, please let me know!

**Comment by mattjj** [comment]
@sschoenholz I lied to you yesterday: you are clearly correct that we don't have a masking rule for reshape! I thought it was in there because [we have a shapecheck rule](https://github.com/google/jax/blob/8c4a938cfa8aec11dcb67660313a25acb9d58fce/tests/masking_test.py#L133).

**Comment by hawkinsp** [comment]
This issue is stale, because `jax.mask` was dropped.
